### PR TITLE
add java solution usaco-673

### PR DESCRIPTION
**Pull request by 876pol**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-673.mdx [hunk 1]
```diff
@@ -92,4 +92,73 @@ int main() {
...
 ```
 </CPPSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class Solution {
```
Comment (SansPapyrus683) [inline solutions/usaco-673.mdx:102]:
```suggestion
public class Team {
```

File: solutions/usaco-673.mdx [hunk 2]
```diff
@@ -92,4 +92,73 @@ int main() {
...
+					dp[i + 1][j + 1][s] += dp[i][j + 1][s];
+					dp[i + 1][j + 1][s] -= dp[i][j][s];
+
+					if (fj[i] > fp[j]) {
+						dp[i + 1][j + 1][s] += dp[i][j][s - 1];
+					}
+
+					// prevent negative results
+					dp[i + 1][j + 1][s] += 1000000009;
+					dp[i + 1][j + 1][s] %= 1000000009;
```
Comment (SansPapyrus683) [inline solutions/usaco-673.mdx:151]: put the mod in a constant called MOD

File: solutions/usaco-673.mdx [hunk 3]
```diff
@@ -92,4 +92,73 @@ int main() {
...
+				}
+			}
+		}
+		io.println(dp[n][m][k]);
+
+		io.close();
```
Comment (SansPapyrus683) [inline solutions/usaco-673.mdx:157]:
```suggestion

		io.println(dp[n][m][k]);
		io.close();
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
lil stuff

**Review by SansPapyrus683 — APPROVED** [review]
image

**Review by envyaims — APPROVED** [review]
ok lol

**Pull request merged**